detect_test ivt fails due to wrong reason silently (xfail masked)

## Current Behavior

I checked xfails for their correctness, they can mask unexpected behaviors. #532 concerns only I-DT but was applied to a test for I-VT. 

in the snippet below

https://github.com/aeye-lab/pymovements/blob/e6c074e632f5dfefcfe8717cff7b69adb954b9f9/tests/unit/gaze/detect_test.py#L362-L370

the offset of the expected dataframe is 1099, while the time column is only 1000-1010, so it is impossible for an offset time to be 1099. thus the wrong implementation of a test was masked by 

## Minimum acceptance criteria

test passes

## Failure Information (for bugs)
Please help provide information about the failure if this is a bug.
If it is not a bug, please remove the rest of this template.

### Steps to Reproduce
apply diff

```diff
diff --git a/tests/unit/gaze/detect_test.py b/tests/unit/gaze/detect_test.py
index fb77b8c..b1f3ddc 100644
--- a/tests/unit/gaze/detect_test.py
+++ b/tests/unit/gaze/detect_test.py
@@ -370,7 +370,6 @@ from pymovements.synthetic import step_function
                 offsets=[1099],
             ),
             id='ivt_constant_position_single_fixation_with_timesteps_float',
-            marks=pytest.mark.xfail(reason='#532'),
         ),

         pytest.param(
```

run 
```console
pytest tests/unit/gaze/detect_test.py
```

## Context

* Project Version / Commit: dev
* Operating System: linux

### Failure Logs


```console
E       AssertionError: DataFrames are different (value mismatch for column 'offset')
E       [left]:  [1009]
E       [right]: [1099]

args       = (shape: (1, 4)
┌──────────┬───────┬────────┬──────────┐
│ name     ┆ onset ┆ offset ┆ duration │
│ ---      ┆ ---   ┆ ...═══════╪═══════╪════════╪══════════╡
│ fixation ┆ 1000  ┆ 1099   ┆ 99       │
└──────────┴───────┴────────┴──────────┘)
function   = <function assert_frame_equal at 0x15523df3ea60>
kwargs     = {'check_row_order': False}
new_name   = 'check_dtypes'
old_name   = 'check_dtype'
version    = '0.20.31'

../../../.local/lib/python3.9/site-packages/polars/_utils/deprecation.py:92: AssertionError
============================================================================================ short test summary info ============================================================================================
XFAIL tests/unit/gaze/detect_test.py::test_gaze_detect[idt_constant_position_single_fixation_with_timesteps_float] - #532
FAILED tests/unit/gaze/detect_test.py::test_gaze_detect[ivt_constant_position_single_fixation_with_timesteps_float] - AssertionError: DataFrames are different (value mismatch for column 'offset')
==================================================================================== 1 failed, 55 passed, 1 xfailed in 2.85s ====================================================================================
```

## Checklist

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed
- [x] I have provided sufficient information for the team

--- a/tests/unit/gaze/detect_test.py
+++ b/tests/unit/gaze/detect_test.py
@@ -359,8 +359,8 @@ from pymovements.synthetic import step_function
                 'minimum_duration': 2,
             },
             pm.gaze.from_numpy(
-                time=np.arange(1000, 1010, 0.1, dtype=float),
-                velocity=step_function(length=100, steps=[0], values=[(0, 0)]),
+                time=np.arange(1000, 1100, 0.1, dtype=float),
+                velocity=step_function(length=1000, steps=[0], values=[(0, 0)]),
                 orient='row',
                 experiment=pm.Experiment(1024, 768, 38, 30, 60, 'center', 1000),
             ),
@@ -370,7 +370,6 @@ from pymovements.synthetic import step_function
                 offsets=[1099],
             ),
             id='ivt_constant_position_single_fixation_with_timesteps_float',
-            marks=pytest.mark.xfail(reason='#532'),
         ),
 
         pytest.param(

